Date picker i18n

yjbanov opened the pull request:
Fixes #12185. This version pull in all localizations provided by `package:intl`. I'm working on a separate PR that reduces `package:intl` to only the list of locales supported by Flutter.

yjbanov committed:
formatYear

yjbanov committed:
localize date picker

yjbanov committed:
tests

yjbanov committed:
clean-ups

Hixie commented on packages/flutter/lib/src/material/date_picker.dart:
```diff
@@ -301,18 +321,64 @@ class DayPicker extends StatelessWidget {
+    final int weekDayFromMonday = new DateTime(year, month).weekday - 1;
+    // 0-based day of week, with 0 representing Sunday.
+    final int firstDayOfWeekFromSunday = localizations.firstDayOfWeekIndex;
+    // firstDayOfWeekFromSunday recomputed to be Sunday-based
```
Monday-based?

yjbanov replied:
LOL. See how confusing this is? 😄  Done. Good catch!

Hixie commented on packages/flutter/lib/src/material/date_picker.dart:
```diff
@@ -902,7 +978,31 @@ Future<DateTime> showDatePicker({
+      textDirection: textDirection,
+      child: child,
+    );
+  }
```
if you provide both, the way you have this built the locale will override the directionality. Presumably if both are given, the directionality should be on the inside.

yjbanov replied:
Done, also actually used the `child` in the dialog! Also added tests.

Hixie commented:
Can we fix https://github.com/flutter/flutter/issues/12050 first? Or at least next? I would rather we didn't keep adding dependencies on `intl`.

Hixie commented on packages/flutter/test/material/date_picker_test.dart:
```diff
@@ -4,7 +4,9 @@
 import 'package:flutter/material.dart';
 import 'package:flutter_test/flutter_test.dart';
-import 'package:intl/intl.dart';
+import 'package:intl/intl.dart' hide TextDirection;
```
as intl

yjbanov replied:
Actually, removed this import altogether. It was used only in one place, which can use `MaterialLocalizations` for this purpose.

Hixie commented:
![LGTM](http://junkyard.damowmow.com/577)

yjbanov committed:
address comments

Hixie commented:
This had a bigger-than-usual negative impact on the aot_snapshot_size benchmarks.

yjbanov commented:
> This had a bigger-than-usual negative impact on the aot_snapshot_size benchmarks.

This is partly because we added localized dates, and partly because we added localized dates for _all locales in the world_. https://github.com/flutter/flutter/pull/12466 partially addresses the issue.

Here are the numbers for building `complex_layout`:

`aot_snapshot_size_isolate`:

Before this PR: 1832972 bytes
After this PR: 1941640 bytes
After #12466: 1867322 bytes

`aot_snapshot_size_rodata`:

Before this PR: 1758568 bytes
After this PR: 1892776 bytes
After #12466: 1794616
